test/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/initialgainorloss/InitialGainOrLossControllerSpec.scala

```diff
@@ -0,0 +1,272 @@
+  implicit lazy val messages: Messages = MessagesImpl(Lang("en"), messagesApi)
+  val initialGainOrLosses              = sample[AmountInPence].copy(value = 2)
+
+  def sessionWithState(
+    answers: CompleteInitialGainOrLossAnswers,
+    disposalDate: DisposalDate
+  ): (SessionData, FillingOutReturn) = {
+    val journey = sample[FillingOutReturn].copy(
+      draftReturn = sample[DraftReturn].copy(
+        initialGainOrLossAnswers = Some(answers)
+      )
+    )
+
+    SessionData.empty.copy(journeyStatus = Some(journey)) -> journey
+  }
+
+  val allQuestionsAnswered: CompleteInitialGainOrLossAnswers = CompleteInitialGainOrLossAnswers(
+    initialGainOrLosses
+  )
+
+  val (session, journey) = sessionWithState(allQuestionsAnswered, sample[DisposalDate])
+  val updatedDraftReturn =
+    journey.draftReturn.copy(initialGainOrLossAnswers = Some(CompleteInitialGainOrLossAnswers(AmountInPence(0))))
+  val updatedSession = session.copy(journeyStatus = Some(journey.copy(draftReturn = updatedDraftReturn)))
+
+  "InitialGainOrLossController" should {
+    "show display" when {
+      def performAction(): Future[Result] = controller.enterInitialGainOrLoss()(FakeRequest())
+      "display a page returning 200" in {
+
+        val fillingOutReturn = sample[FillingOutReturn]
+
+        inSequence {
+          mockAuthWithNoRetrievals()
+          mockGetSession(
+            SessionData.empty.copy(
+              journeyStatus = Some(fillingOutReturn)
+            )
+          )
+        }
+
+        val result: Future[Result] = performAction()
+        status(result) shouldBe OK
+      }
+
+      "have proper contents" in {
+        val fillingOutReturn = sample[FillingOutReturn]
+
+        inSequence {
+          mockAuthWithNoRetrievals()
+          mockGetSession(
+            SessionData.empty.copy(
+              journeyStatus = Some(fillingOutReturn)
+            )
+          )
+        }
+
+        val result: Future[Result] = performAction()
+
+        contentAsString(result) should include("Initial gain or loss")
+        contentAsString(result) should include("Did you make an initial gain or loss?")
+      }
+
+      "display the page" in {
+        val fillingOutReturn = sample[FillingOutReturn]
+        inSequence {
+          mockAuthWithNoRetrievals()
+          mockGetSession(
+            SessionData.empty.copy(
+              journeyStatus = Some(fillingOutReturn)
+            )
+          )
+        }
+        checkPageIsDisplayed(
+          performAction(),
+          messageFromMessageKey(
+            "initialGainOrLoss.title"
+          ), { doc =>
+            doc.select("#back").attr("href") shouldBe returns.routes.TaskListController.taskList().url
+            doc.select("#content > article > form").attr("action") shouldBe routes.InitialGainOrLossController
+              .submitInitialGainOrLoss()
+              .url
+          }
+        )
+      }
+
+    }
+  }
+
+  "show an error page" when {
```

Comment:
there's some missing coverage here, e.g. when submits are successful what happens and coverage for the check you answers page

Reply:
Done.